plugin/environment.go

```diff
@@ -488,6 +498,50 @@ func (env *Environment) UnpackWebappBundle(id string) (*model.Manifest, error) {
 	return manifest, nil
 }
 
+func patchReactDOM(b []byte) ([]byte, bool) {
+	s := string(b)
+	nameIndex := strings.Index(s, "react-dom.production.min.js")
+
+	if nameIndex == -1 {
+		return b, false
+	}
+
+	beginning := strings.LastIndex(s[:nameIndex], "{")
+	var end int
+
+	argDefBeginning := strings.LastIndex(s[:beginning], "function") + 9
+	argDefEnd := strings.LastIndex(s[:beginning], ")") - 1
+	argsNames := strings.Split(s[argDefBeginning:argDefEnd], ",")
+	if len(argsNames) != 3 {
+		return b, false
+	}
+
+	exportsArgName := strings.TrimSpace(argsNames[1])
```

Comment:
Here's the GitHub plugin's compiled bundle for reference: [main.js](https://github.com/mattermost/mattermost-server/files/9923522/main.js.txt)

The relevant contents of the bundle are an array of functions that each define a package's exports. Here's what the `ReactDOM` part looks like:

```js
function(e,t,n){"use strict";
/** @license React v16.8.6
 * react-dom.production.min.js
 *
 ```

followed by some more comments, and then all of its exports. This PR's code finds where this function ends, and injects the `Object.assign(%s, window.ReactDOM)` statement at the end of the function, which effectively overrides the previously defined exports. The `exports` object is the second parameter, in this case `t`. The code looks for this arg name, and uses it in the `Object.assign` statement.

I've tried researching the anatomy of a webpack bundle, but since it's an implementation detail, there isn't a lot documented on the surface. Overriding the properties on the `exports` object seems like a reasonable approach here, though I'm wondering if there's something more appropriate that should be done in this hacky fix.